### PR TITLE
Fixed event: starting shared-components

### PR DESCRIPTION
**Pull request by whimsicallyson**
## Links
* https://fixed-event.glitch.me/
* https://app.zenhub.com/workspaces/community-5d07d1359dee6059a9688176/issues/fogcreek/glitch-community-work/363
https://app.zenhub.com/workspaces/community-5d07d1359dee6059a9688176/issues/fogcreek/glitch-community-work/365
partially https://app.zenhub.com/workspaces/community-5d07d1359dee6059a9688176/issues/fogcreek/glitch-community-work/360

## Changes:
* We shouldn't have to use the Emoji or Button code from ~community anymore (in fact, they're already deleted in this remix). I'm still working on AnimationContainer in one spot though, so its not removed.

## How To Test:
* Click around, look for anything broken or inconsistent. There will be some slight changes in font size, spacing, colors, because we're condensing down to a smaller number of total options, but nothing should look actively *wrong*. If it does, point it out!

## Feedback I'm looking for:
Is anything broken? Does it all seem ok still?

## Things left to do before deploying:
- Disabled button styling should work properly once https://github.com/FogCreek/shared-components/pull/38 is merged in.

ps @modernserf you may be interested in this, as a reviewer or just observer.

### CODE REVIEW

File: src/components/create-team-pop/styles.styl [hunk 1]
```diff
@@ -6,4 +6,10 @@
   font-family: monospace
   font-size: 13px
   color: tertiary
-  margin: 12px 0
\ No newline at end of file
+  margin: 12px 0
+  
+img.emoji
+  height: 1.1em
```
Comment (clottman) [inline src/components/create-team-pop/styles.styl]: Is this intended to be a different size than the other emoji? Or can we standardize it to 1.3em?

File: src/components/pagination-controller/index.js [hunk 2]
```diff
@@ -99,18 +99,18 @@ function PaginationController({ enabled, items, itemsPerPage, fetchDataOptimisti
...
             <div data-cy="page-numbers" className={styles.pageNumbers}>
               {state.page} / {numPages}
             </div>
-            <Button ref={nextButtonRef} type="tertiary" disabled={state.page === numPages} onClick={onNextButtonClick}>
+            <Button ref={nextButtonRef} variant="secondary" disabled={state.page === numPages} onClick={onNextButtonClick}>
               <Image alt="Next" className={classNames(styles.paginationArrow, styles.next)} src={arrow} />
             </Button>
           </div>
-          <Button data-cy="show-all" type="tertiary" onClick={() => dispatchState({ type: 'expand' })}>
+          <Button data-cy="show-all" variant="secondary" onClick={() => dispatchState({ type: 'expand' })}>
```
Comment (clottman) [inline src/components/pagination-controller/index.js]: image
The badge is misaligned here

File: src/components/footer/index.js [hunk 3]
```diff
@@ -21,33 +22,35 @@ const PlatformsIcon = () => (
...
-      Legal Stuff
-      <Emoji name="scales" />
-    </FooterLink>
-    <FooterLink className={styles.teams} href="/teams" track="platforms">
-      <PlatformsIcon />
-      Glitch Teams
-    </FooterLink>
-  </footer>
-);
+function Footer() {
```
Comment (clottman) [inline src/components/footer/index.js]: Why the change to `{ return (..) }` over just ` => (..)`? Can we put it back to the simple way?

File: src/components/header/new-project-pop.js [hunk 4]
```diff
@@ -61,7 +61,7 @@ const NewProjectPop = ({ projects }) => (
       )}
     </PopoverSection>
     <PopoverActions type="secondary">
-      <Button size="small" type="tertiary" onClick={importGitRepo} matchBackground>
+      <Button size="small" type="secondary" onClick={importGitRepo} matchBackground>
```
Comment (clottman) [inline src/components/header/new-project-pop.js]: `variant`?

File: src/components/images/emoji.js~c52f1d17f7c3fdb4d63f7904fcd82da2f952e10c [hunk 5]
```diff
@@ -0,0 +1,85 @@
+import React from 'react';
```
Comment (clottman) [inline src/components/images/emoji.js~c52f1d17f7c3fdb4d63f7904fcd82da2f952e10c]: What's this file? The name is really wonky.

File: src/components/new-stuff/new-stuff-article.js [hunk 6]
```diff
@@ -3,16 +3,18 @@ import PropTypes from 'prop-types';
...
 
 import styles from './new-stuff-article.styl';
 
-const NewStuffArticle = ({ title, body, link }) => (
+const NewStuffArticle = ({ title, body, image, imageAlt, link }) => (
   <article className={styles.article}>
     <h2 className={styles.title}>{title}</h2>
     <div className={styles.body}>
       <Markdown>{body}</Markdown>
+      {image ? <Image src={image} alt={imageAlt || ''} /> : '' }
```
Comment (clottman) [inline src/components/new-stuff/new-stuff-article.js]: This can be simplified to `image && <Image src...`

File: src/components/popover/index.js [hunk 7]
```diff
@@ -80,6 +81,7 @@ export const PopoverWithButton = ({ buttonProps, buttonText, children: renderChi
             <div className={styles.buttonWrap}>
               <Button {...buttonProps} ref={buttonRef} onClick={popoverProps.togglePopover}>
                 {buttonText}
+                {buttonProps.emoji ? <Icon icon={buttonProps.emoji} className={styles.emoji} /> : ''}
```
Comment (clottman) [inline src/components/popover/index.js]: lets shorten this up too: `buttonProps.emoji && <Icon...`

File: src/components/popover/index.js [hunk 8]
```diff
@@ -139,8 +141,8 @@ PopoverMenu.defaultProps = {
 // Use with PopoverMenu so that popover can correctly adjust to fit the content
 export const PopoverMenuButton = ({ label, emoji, onClick }) => (
   <div className={styles.menuButtonWrap}>
-    <Button size="small" type="tertiary" emoji={emoji} onClick={onClick}>
-      <div className={styles.popoverButtonContent}>{label}</div>
+    <Button size="small" variant="secondary" onClick={onClick}>
+      <div className={styles.popoverButtonContent}>{label} <Icon icon={emoji} /></div>
```
Comment (clottman) [inline src/components/popover/index.js]: Is this wrapper div still necessary or can we move the class to be on Button? I think it might be a remnant of when we weren't allowing a `className` prop on `Button`

File: src/components/popover/styles.styl [hunk 9]
```diff
@@ -109,3 +109,9 @@
 
 .menuButtonWrap + .menuButtonWrap
   margin-top: 8px
+
+img.emoji
+  height: 1.1em
```
Comment (clottman) [inline src/components/popover/styles.styl]: Another short emoji - intentional?

File: src/components/project/featured-project.styl [hunk 10]
```diff
@@ -20,4 +20,10 @@
 
 .bookmarkButtonContainer
   display: block
-  margin: -10px 0px 0 0
\ No newline at end of file
+  margin: -10px 0px 0 0
+  
+img.emoji
+  height: 1.1em
```
Comment (clottman) [inline src/components/project/featured-project.styl]: We're setting these classes all over the place now with nearly identical CSS - (either 1.1 or 1.3 em). What do you think about making our own component that wraps the Icon and adds styles to make it one of these standard styles, or maybe adding this to the shared components repo? 
@modernserf

File: src/components/project/project-actions.js [hunk 11]
```diff
@@ -22,17 +22,17 @@ export const EditButton = ({ name, isMember, size }) => (
 EditButton.propTypes = {
   name: PropTypes.string.isRequired,
   isMember: PropTypes.bool,
-  size: PropTypes.oneOf(SIZES),
```
Comment (clottman) [inline src/components/project/project-actions.js]: Why do we need this change? Can we export the acceptable button sizes from the shared components library so we can still have prop type checking as strict as we previously did? 
@modernserf

File: src/components/project/project-options-pop.js [hunk 12]
```diff
@@ -18,7 +19,7 @@ const PopoverMenuItems = ({ children }) =>
   children.map(
     (group, i) =>
       group.some((item) => item.onClick) && (
-        <PopoverActions key={i} type={group.some((item) => item.dangerZone) ? 'dangerZone' : undefined}>
+        <PopoverActions key={i} variant={group.some((item) => item.dangerZone) ? 'warning' : undefined}>
```
Comment (clottman) [inline src/components/project/project-options-pop.js]: Did you mean to change this on `PopoverActions`?

Comment (clottman) [reply src/components/project/project-options-pop.js]: I think this broke it actually 
before: 
on glitch.com

after: 
image

File: src/components/recent-projects/index.js [hunk 13]
```diff
@@ -34,8 +34,8 @@ const ClearSession = ({ clearUser }) => {
 
   return (
     <div className={styles.clearSession}>
-      <Button onClick={clickClearSession} size="small" type="dangerZone" emoji="balloon">
-        Clear Session
+      <Button onClick={clickClearSession} size="small">
```
Comment (clottman) [inline src/components/recent-projects/index.js]: Did we lose the `dangerZone`ness here?

File: src/components/sign-in/password-login.js [hunk 14]
```diff
@@ -0,0 +1,86 @@
+import React, { useState } from 'react';
```
Comment (clottman) [inline src/components/sign-in/password-login.js]: I think this was supposed to be deleted, not added, in a recent PR

File: src/components/user-options-pop/index.js [hunk 15]
```diff
@@ -43,12 +42,13 @@ const TeamList = ({ teams, showCreateTeam }) => {
...
           <Button
+            as="a"
             href={getTeamLink(team)}
             size="small"
-            type="tertiary"
-            image={<TeamAvatar team={team} size="small" tiny hideTooltip />}
+            variant="secondary"
           >
             {team.name}
+            <TeamAvatar team={team} size="small" style={{ marginLeft: `${5}px` }} tiny hideTooltip />
```
Comment (clottman) [inline src/components/user-options-pop/index.js]: Another weird interpolation. `5px` should be sufficient.

File: src/curated/pupdates.json [hunk 16]
```diff
@@ -0,0 +1 @@
...ow live on your profile","body":"Have a project you're particularly proud of? Feature it to embed the app directly at the top of your profile.<br/><br/>You can feature one project at a time – and change or remove your featured project as often as you'd like.","image":"https://cdn.glitch.com/1b4c205a-e6b6-4ad9-ae8a-1e511e611105%2Ffeature.png?1548796979652"},{"id":2,"title":"Use your email to get Glitchy","body":"Don't have Facebook or GitHub, but still want to use Glitch? Want to code at the library without remembering all your logins? Now you can sign into Glitch with just your email.<br/><br/>Choose Sign in with Email to get a magic link and temporary sign-in code made just for you and delivered straight to your inbox.","image":"https://cdn.glitch.com/69bf2d27-a029-4632-86a4-a722974778ae%2Femail-sign-in.gif?1548196863943"},{"id":1,"title":"Get organized with Collections","body":"Add any project to a collection and show off your favorite things, group all your related projects, collect precious memories, etc. Head over to your profile or team page to create and view your collections.","image":"https://cdn.glitch.com/e9551d04-9c48-4dbb-ab1b-04eca25b9652%2Fimage.png?1547136745894"}]}
```
Comment (clottman) [inline src/curated/pupdates.json]: Reminder to check that these are the right pupdates

File: src/presenters/pages/home-v2/index.js [hunk 17]
```diff
@@ -192,45 +189,36 @@ const UnifiedStories = ({ content: { hed, dek, featuredImage, featuredImageDescr
...
-                  Read More on Culture <Arrow />
-                </Button>
-              </div>
-            </>
-          </LazyLoader>
-        </div>
-      </HomeSection>
-    )}
-  </VisibilityContainer>
+  <HomeSection id="culture-zine" className={styles.cultureZine}>
```
Comment (clottman) [inline src/presenters/pages/home-v2/index.js]: Haven't seen these changes come through, maybe double check that they are supposed to be there.

File: src/presenters/pages/home-v2/index.js [hunk 18]
```diff
@@ -292,6 +281,7 @@ export const Home = ({ data, loggedIn, hasProjects }) => (
 export const HomePreview = () => {
   const api = useAPI();
   const { origin, ZINE_POSTS } = useGlobals();
+  console.log('HomePreview', origin);
```
Comment (clottman) [inline src/presenters/pages/home-v2/index.js]: Probably remove this?

File: src/presenters/pages/user.styl [hunk 19]
```diff
@@ -1,2 +1,7 @@
 @import  "../../components/constants"
 
+img.emoji
+  height: 20px
```
Comment (clottman) [inline src/presenters/pages/user.styl]: Why is this one in px instead of em?

**Comment by clottman** [comment]
create button disabled
The create button is disabled in the screenshot above, but it's not visually distinct from a working button. @modernserf

**Comment by clottman** [comment]
In the sign-in pop, it looks like we're missing a space between the team name and the team avatar now. 
New: 
sign in pop team list on this remix

on prod: 
sign in pop team list on glitch.com

File: src/components/user-options-pop/index.js [hunk 20]
```diff
@@ -129,30 +129,30 @@ Are you sure you want to sign out?`)
       <PopoverInfo>
         {(canBecomeSuperUser || !!superUserFeature) && (
           <div className={styles.buttonWrap}>
-            <CheckboxButton value={!!superUserFeature} onChange={toggleSuperUser} type="tertiary">
+            <CheckboxButton value={!!superUserFeature} onChange={toggleSuperUser} variant="secondary">
```
Comment (clottman) [inline src/components/user-options-pop/index.js]: Does checkbox button have size=small too? 
This button is bigger than the other buttons in this menu now 
image

**Comment by clottman** [comment]
The hover state of the team buttons in the sign-in pop underlines the team name now - was that an intentional change? 
image

**Comment by clottman** [comment]
Sometimes the buttons get too big 
https://fixed-event.glitch.me/@cassey/weddings

They used to wrap: 
image

https://fixed-event.glitch.me/@cassey/weddings
and https://glitch.com/@cassey/weddings

**Comment by clottman** [comment]
When I've added a user to a team, the button to remove them is a different size than the other button 
image

**Comment by clottman** [comment]
This is really subtle because it usually goes fast, but the delete team button (when you actually delete a team) has a loader that's too big for the button. So it makes the button get bigger and whereas before, the button was small enough to still fit in the pop, now the button overflows the popover. 

image
See a video of this flow [on the drive](https://drive.google.com/open?id=1xB_TkL-7oQegDtMqAufihI1gTIJOR_GW). Turn off network or throttle your connection while deleting a team to see it more clearly for yourself.

It looks like a space also disappeared between the project name and the bomb icon.

File: src/components/containers/expander.js [hunk 21]
```diff
@@ -54,7 +53,7 @@ const Expander = ({ children, height, minSlide }) => {
       {expandState !== 'complete' && aboveLimit && (
         <div className={styles.expanderMask}>
           {expandState !== 'expanding' && (
-            <Button size="small" type="tertiary" onClick={expand}>
+            <Button size="small" variety="secondary" onClick={expand}>
```
[the remark this answers is not in the document]
Comment (whimsicallyson) [reply src/components/containers/expander.js]: yup!

File: src/components/footer/index.js (hunk 3 above)
Comment (whimsicallyson) [reply src/components/footer/index.js]: yeah, that's from when I was trying to use styled components to fix the emoji and I never changed it back. fixed!

File: src/components/images/emoji.js~c52f1d17f7c3fdb4d63f7904fcd82da2f952e10c (hunk 5 above)
Comment (whimsicallyson) [reply src/components/images/emoji.js~c52f1d17f7c3fdb4d63f7904fcd82da2f952e10c]: huh. definitely not in there on purpose.

File: src/curated/pupdates.json (hunk 16 above)
Comment (whimsicallyson) [reply src/curated/pupdates.json]: confirmed!

**Comment by whimsicallyson** [comment]
> Sometimes the buttons get too big
> https://fixed-event.glitch.me/@cassey/weddings
> 
> They used to wrap:
> image
> 
> https://fixed-event.glitch.me/@cassey/weddings
> and https://glitch.com/@cassey/weddings

@modernserf this would be fixed with `white-space: normal` on buttons, but you currently have them set to `nowrap`. thoughts on changing it? (since this is something that can happen nearly anywhere, I don't think it makes sense to override for certain buttons)

**Comment by modernserf** [comment]
hmm -- i think the `nowrap` is needed for buttons in popovers, since the width of the popovers is set by the size of the buttons. That said, I think it probably makes more sense to have the popovers add the nowrap style instead of you overriding it here.

File: src/components/user-options-pop/index.js (hunk 15 above)
Comment (whimsicallyson) [reply src/components/user-options-pop/index.js]: changed, also fixes the lack of padding for team avatars in the user options popover pictured below.

File: src/components/buttons/sign-in-button.js [hunk 22]
```diff
@@ -65,8 +64,8 @@ const SignInButton = ({ companyName, onClick, short }) => {
 
   return (
     <div style={{ marginBottom: '10px' }}>
-      <Button href={href} onClick={onClick} size="small" emoji={emoji}>
-        {short ? name : `Sign in with ${name}`}
+      <Button href={href} onClick={onClick} size="small">
+        {short ? name : `Sign in with ${name}`} <Icon style={{ height: `${14}px`, width: `${14}px`, verticalAlign: 'middle' }} icon={emoji} />
```
[the remark this answers is not in the document]
Comment (whimsicallyson) [reply src/components/buttons/sign-in-button.js]: updated

File: src/components/collection/add-collection-project-pop.js [hunk 23]
```diff
@@ -99,7 +99,7 @@ function AddCollectionProjectPop({ collection, togglePopover, addProjectToCollec
       {status === 'ready' && excludingExactMatch && (
         <PopoverInfo>
           <InfoDescription>
-            {parsedQuery} is already in this collection <Emoji name="sparkles" />
+            {parsedQuery} is already in this collection <Icon style={{ height: `${1.3}em`, width: `${1.3}em` }} icon="sparkles" />
```
[the remark this answers is not in the document]
Comment (whimsicallyson) [reply src/components/collection/add-collection-project-pop.js]: also updated

File: src/components/create-team-pop/styles.styl (hunk 1 above)
Comment (whimsicallyson) [reply src/components/create-team-pop/styles.styl]: this is from when I was trying to get everything matched exactly to prod, before talking with Justin and determining that was not exactly the point of componentization :) standardized to 1.3 with everything else!

File: src/components/pagination-controller/index.js (hunk 2 above)
Comment (whimsicallyson) [reply src/components/pagination-controller/index.js]: this is fixed by using the Badge shared-component, so I'm gonna do that even though implementing it in general will be a separate PR

File: src/components/popover/index.js (hunk 8 above)
Comment (whimsicallyson) [reply src/components/popover/index.js]: doesn't seem necessary!

File: src/components/sign-in/password-login.js (hunk 14 above)
Comment (whimsicallyson) [reply src/components/sign-in/password-login.js]: good catch!

File: src/presenters/pages/home-v2/index.js (hunk 17 above)
Comment (whimsicallyson) [reply src/presenters/pages/home-v2/index.js]: updated to match prod.

File: src/presenters/pages/user.styl (hunk 19 above)
Comment (whimsicallyson) [reply src/presenters/pages/user.styl]: not exactly sure, standardized!

File: src/components/project/featured-project.styl (hunk 10 above)
Comment (whimsicallyson) [reply src/components/project/featured-project.styl]: > This is really subtle because it usually goes fast, but the delete team button (when you actually delete a team) has a loader that's too big for the button. So it makes the button get bigger and whereas before, the button was small enough to still fit in the pop, now the button overflows the popover.
> 
> image
> See a video of this flow [on the drive](https://drive.google.com/open?id=1xB_TkL-7oQegDtMqAufihI1gTIJOR_GW). Turn off network or throttle your connection while deleting a team to see it more clearly for yourself.
> 
> It looks like a space also disappeared between the project name and the bomb icon.

fixed the spacing between text and icon, the loader issue should be fixed with the change of `white-space: normal` to the button component

**Comment by whimsicallyson** [comment]
> This is really subtle because it usually goes fast, but the delete team button (when you actually delete a team) has a loader that's too big for the button. So it makes the button get bigger and whereas before, the button was small enough to still fit in the pop, now the button overflows the popover.
> 
> image
> See a video of this flow [on the drive](https://drive.google.com/open?id=1xB_TkL-7oQegDtMqAufihI1gTIJOR_GW). Turn off network or throttle your connection while deleting a team to see it more clearly for yourself.
> 
> It looks like a space also disappeared between the project name and the bomb icon.

fixed the spacing between text and icon, the loader issue should be fixed with the change of white-space: normal to the button component

File: src/components/project/featured-project.styl (hunk 10 above)
Comment (whimsicallyson) [reply src/components/project/featured-project.styl]: moved to a global style that you can import as needed--still not as elegant as I'd like, but better than literally copy-pasting the code everywhere.

**Comment by whimsicallyson** [comment]
I believe that between the changes I've made, and the changes in https://github.com/FogCreek/shared-components/pull/39 & https://github.com/FogCreek/shared-components/pull/40, everything should be addressed?

**Comment by ttseng** [comment]
I notice that vertical alignment in buttons with emojis / images seems off, for example in the user options pop:

image
(current alignment on left, "fixed" alignment on right)

You can notice it the most for buttons like 'Create Team' where there is less spacing below the text + emoji than above

I think this can be fixed by adjusting the button padding from 
`padding: 0.375em 0.5em 0.1875em`
to
`padding: 0.3em 0.5em 0.285em`

and also making the emoji + image sizing consistent (right now, the tiny avatars have a fixed width and height of 16px, but emojis are set at 1.3em)

**Comment by ttseng** [comment]
The padding / alignment is currently a bit off with `coverWrap` and `projectWrap` - notice how they are taller on `fixed-event` than on `production.

image

**Comment by whimsicallyson** [comment]
emoji alignment/padding should be better now, as well as the padding/alignment on `projectWrap`/`coverWrap`. The active color on buttons is baked into the shared components lib, so I'm hesitant to override that unless its a special case (thoughts there, @modernserf ?)

**Comment by modernserf** [comment]
@ttseng 
here's the discussion we had on using dark blue as the `:active` state on buttons: https://github.com/FogCreek/shared-components/pull/32#discussion_r313944031

I think we need _some_ sort of active state on buttons for a11y purposes. It might make more sense to have it match the `:hover` state instead of the dark blue, though I'm not sure if there's enough contrast from the inactive state for it to be accessible -- maybe @clottman can weigh in on this.

**Comment by modernserf** [comment]
unrelated -- it seems like we've lost the link on the "Read more on Culture" button

File: src/components/buttons/story.js [hunk 24]
```diff
@@ -2,69 +2,10 @@ import React from 'react';
 import { storiesOf } from '@storybook/react';
```
Comment (clottman) [inline src/components/buttons/story.js]: This file shouldn't exist any more if you're up to date with master - can you merge and make sure all the storybook files are still gone? I worry that since you touched them they might sneak back in

File: src/components/collection/collection-item-small.js [hunk 25]
```diff
@@ -50,7 +49,7 @@ const CollectionItemSmall = ({ collection, showCurator }) => (
           </div>
           <div className={styles.collectionNameWrap}>
             <div className={styles.itemButtonWrap}>
-              <Button decorative image={collection.private ? <PrivateIcon inButton isPrivate /> : null} imagePosition="left">
+              <Button as="span" image={collection.private ? <Icon icon="private" /> : null} imagePosition="left">
```
[the remark this answers is not in the document]
Comment (clottman) [reply src/components/collection/collection-item-small.js]: @modernserf @whimsicallyson did this get resolved or do we need to create another ticket for it?

File: src/components/containers/preview-container.styl [hunk 26]
```diff
@@ -13,3 +13,7 @@
   flex: 1 0 auto
   font-size: 16px
   font-weight: bold
+
+dialog
+  top: 25%!important
```
Comment (clottman) [inline src/components/containers/preview-container.styl]: Does this work with no space between `25%` and `!important`?

File: src/components/project/project-actions.js (hunk 11 above)
Comment (clottman) [reply src/components/project/project-actions.js]: bump!

File: stories/index.js [hunk 27]
```diff
@@ -2,14 +2,11 @@ import React from 'react';
 import { mapValues, sumBy, memoize } from 'lodash';
```
Comment (clottman) [inline stories/index.js]: Another story file that shouldn't exist any more

File: src/components/collection/collection-item-small.js (hunk 25 above)
[the remark this answers is not in the document]
Comment (whimsicallyson) [reply src/components/collection/collection-item-small.js]: created https://app.zenhub.com/workspaces/community-5d07d1359dee6059a9688176/issues/fogcreek/shared-components/42

File: src/components/containers/preview-container.styl (hunk 26 above)
Comment (whimsicallyson) [reply src/components/containers/preview-container.styl]: yes but I fixed it anyhow :)
